packages/babel-types/src/converters/toIdentifier.ts

```diff
@@ -1,10 +1,14 @@
 import isValidIdentifier from "../validators/isValidIdentifier";
+import { isIdentifierChar } from "@babel/helper-validator-identifier";
 
-export default function toIdentifier(name: string): string {
-  name = name + "";
+export default function toIdentifier(input: string): string {
+  input = input + "";
 
   // replace all non-valid identifiers with dashes
-  name = name.replace(/[^a-zA-Z0-9$_]/g, "-");
+  let name = "";
+  for (const c of input) {
+    name += isIdentifierChar(c.charCodeAt(0)) ? c : "-";
```

Comment:
```suggestion
    name += isIdentifierChar(c.codePointAt(0)) ? c : "-";
```
non-BMP characters will be broken into surrogate pairs which will never be valid `ID_Continue`.